### PR TITLE
Introduce solidity_free in libsolc

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libsolc/libsolc.h [hunk 1]
```diff
@@ -55,6 +55,12 @@ char const* solidity_version() SOLC_NOEXCEPT;
 /// The pointer returned must not be freed by the caller.
 char const* solidity_compile(char const* _input, CStyleReadFileCallback _readCallback) SOLC_NOEXCEPT;
 
+/// Cleans up any allocated memory.
+///
+/// NOTE: the pointers returned by solidity_license, solidity_version and solidity_compile
```
Comment (axic) [inline libsolc/libsolc.h]: Actually license/version is not affected.

**Review by chriseth — CHANGES_REQUESTED** [review]
Please fix the comment.

**Comment by axic** [comment]
Fixed, but are we happy with the naming?

**Comment by chriseth** [comment]
Perhaps something like `relaseBuffer` / `freeBuffer`?

By the way, did you check this actually has an effect in emscripten?

**Comment by axic** [comment]
> By the way, did you check this actually has an effect in emscripten?

Solc-js copies the result from `compile`, so calling this wouldn't have an adverse effect. Haven't checked if emscripten gets any actual benefit from this - it definitely shouldn't break anything. My motivation for this is other use cases where it is statically linked, such as Rust, but it should also help when compiling libsolc to webassembly.

**Comment by axic** [comment]
> Perhaps something like `relaseBuffer` / `freeBuffer`?

I was thinking along the lines this function would "reset" the state, should we have more state in the future. If de not except anything like that then release/free might be better.

**Comment by chriseth** [comment]
`cleanup` is something I would expect to call at the very end, right before shutdown, so either `releaseBuffers` or `reset` would be better in my opinion.

I would have assumed the standard-json interface to carry no state.

**Comment by axic** [comment]
> I would have assumed the standard-json interface to carry no state.

I would hope we wouldn't introduce any state into this API.

**Comment by axic** [comment]
@christianparpart @ekpyron @bit-shift @leonardoalt what do you think, what would be a good naming for this function?

**Comment by leonardoalt** [comment]
I like `free`, especially because it's a C API ;)

**Comment by bshastry** [comment]
I like `solidity_free` since `malloc`/`free` is a familiar C idiom for memory (de)allocation :)

**Comment by axic** [comment]
Renamed to free, though I'm not satisfied with any of the namings :)

**Review by chriseth — DISMISSED** [review]

**Comment by axic** [comment]
Rebased after some other PR changed fuzzer common. Will merge once tests pass.

**Pull request merged**